src/tests/ftest/deployment/network_failure.py

```diff
@@ -0,0 +1,398 @@
+
+        1. Create a pool and a container. Create a container with or without redundancy
+        factor based on container_namespace.
+        2. Take down network interface of one of the engines, say ib0 of rank 0. hsn0 in
+        Aurora.
+        3. Run IOR with given object class.
+        4. Bring up the network interface.
+        5. Restart DAOS with dmg.
+        6. Call dmg pool query -b to find the disabled ranks.
+        7. Call dmg pool reintegrate --rank=<rank> one rank at a time to enable all
+        ranks. Wait for rebuild after calling the command.
+        8. Run IOR again. It should work this time.
+        9. To further verify the system, create another container.
+        10. Run IOR to the new container. Should work.
+
+        Note that I'm not sure about the usefulness of testing different object classes
+        and redundancy factors. We probably have to understand how data are exchanged
+        among the ranks based on the object class.
+
+        Args:
+            ior_namespace (str): Yaml namespace that defines the object class used for
+                IOR.
+            container_namespace (str): Yaml namespace that defines the container
+                redundancy factor.
+        """
+        # 1. Create a pool and a container.
+        self.container = []
+        self.add_pool(namespace="/run/pool_size_ratio_80/*")
+        self.container.append(
+            self.get_container(pool=self.pool, namespace=container_namespace))
+
+        # 2. Take down network interface of one of the engines. Use the first host.
+        errors = []
+        self.network_down_host = self.hostlist_servers[0]
+        self.log.info("network_down_host = %s", self.network_down_host)
+        self.interface = self.params.get(
+            "fabric_iface", "/run/server_config/servers/0/*")
+        self.log.info("interface to update = %s", self.interface)
+
+        if self.test_env == "ci":
+            # wolf
+            self.update_network_interface(
+                interface=self.interface, state="down", host=self.network_down_host,
+                errors=errors)
+        else:
+            # Aurora
+            command = f"sudo ip link set {self.interface} down"
+            self.log.debug("## Call %s on %s", command, self.network_down_host)
+            time.sleep(20)
+
+        # 3. Run IOR with given object class. It should fail.
+        job_num = 1
+        ior_results = {}
+        # IOR will not work, so we'll be waiting for the Mpirun timeout.
+        self.run_ior_report_error(
+            job_num=job_num, results=ior_results, file_name="test_file_1",
+            pool=self.pool, container=self.container[0], namespace=ior_namespace,
+            timeout=10)
+        self.log.info(ior_results)
+
+        # 4. Bring up the network interface.
+        if self.test_env == "ci":
+            # wolf
+            self.update_network_interface(
```

Comment:
(pylint-no-member) Instance of 'NetworkFailureTest' has no 'update_network_interface' member